src/database/UIDatabase.js

```diff
@@ -35,36 +33,53 @@ export class UIDatabase {
     this.database = database;
   }
 
+  ERRORS = {
+    ERROR_IN_TRANSACTION: { code: 'ERROR_IN_TRANSACTION', message: 'Database is in a transaction' },
+    ERROR_NO_PERMISSION: { code: 'ERROR_NO_PERMISSION', message: 'Storage permission not granted' },
+    ERROR_UNKNOWN: { code: 'ERROR_UNKOWN', message: 'ERROR_UNKNOWN' },
+  };
+
   /**
-   * Closes the database, exports the realm file to '/Download/mSupplyMobile\ data' on device
-   * file system. The app will crash if anything tries to access the database while it is closed.
+   * Exports the realm file to '/Download/mSupplyMobile\ data' on device file system.
+   * Ensures there is enough space, the realm exists and requests storage permission,
+   * if required.
    */
-  exportData(filename = 'msupply-mobile-data') {
-    // TO DO: add method to react-native-database.
 
+  async exportData(filename = 'msupply-mobile-data') {
```

Comment:
Wonder if worth putting this and directory path in a constant? e.g.:

```
EXPORT_DIR = '/Download/mSupplyMobile_data'
EXPORT_FILE = 'msupply-mobile-data'
```

Or, if you like objects:

```
PATHS: {
   export_dir: '/Download/mSupplyMobile_data'
   export_file: 'msupply-mobile-data'
}
```

Or some variation of this.

Reply:
5c6bf9ca298ce98bf2b615059c7410aa90aebfc5 cheers